pkg/network/node/sdn_controller.go

```diff
@@ -129,16 +129,21 @@ func (plugin *OsdnNode) SetupSDN() (bool, error) {
 	gwCIDR := fmt.Sprintf("%s/%d", localSubnetGateway, localSubnetMaskLength)
 
 	if err := waitForOVS(ovsDialDefaultNetwork, ovsDialDefaultAddress); err != nil {
-		return false, err
+		return false, nil, err
 	}
 
 	var changed bool
+	var existingPods map[string]podNetworkInfo
 	if err := plugin.alreadySetUp(gwCIDR, clusterNetworkCIDRs); err == nil {
```

Comment:
Do we need to worry about the case where the SDN is partially set up? `alreadySetUp` checks a few things, but it isn't exhaustive.

I don't think so, since a pod would have to completely succeed setup to transition to Running.

Reply:
`alreadySetUp` is meant to distinguish between:

1. the network is correctly set up
2. the network is not set up at all
3. the network was set up, but the networking config has changed since then
4. the network was set up, but openshift-sdn was upgraded and now wants things set up differently

It is not intended to handle "the network was set up, but then the node was hit by cosmic rays" or "the network was set up, but then the admin manually broke things". In that case, reboot.

Reply:
I think it's possible that we can get into the partial state, but pods wouldn't be working before this code gets hit anyway. The first time the SDN runs it would have to crash mid-setup, then the second time would have to set up some pods successfully, but they'd already be broken because we're not fully set up. And then finally the 3rd time it runs it might encounter the problem here.

But we already have that problem because the existing code would kill the pods, then let kubelet restart them, and they'd still be broken because of the partial setup of the bridge. So I dont' think there would be any regression here.